confgenerator/confgenerator_test.go

```diff
@@ -129,39 +125,41 @@ func testGenerateConfsWithValidInput(t *testing.T, platform platformConfig) {
 			// Retrieve the expected golden conf files.
 			expectedMainConfig := readFileContent(t, platform.OS, testName, goldenMainPath, true)
 			expectedParserConfig := readFileContent(t, platform.OS, testName, goldenParserPath, true)
+
 			// Generate the actual conf files.
-			setFilepathJoin(platform.OS)
 			mainConf, parserConf, err := uc.GenerateFluentBitConfigs(platform.defaultLogsDir, platform.defaultStateDir, platform.InfoStat)
-			setFilepathJoin(runtime.GOOS)
 			if err != nil {
-				t.Fatalf("GenerateFluentBitConfigs got %v", err)
+				t.Errorf("GenerateFluentBitConfigs got %v", err)
+			} else {
+				// Compare the expected and actual and error out in case of diff.
+				updateOrCompareGolden(t, testName, platform.OS, expectedMainConfig, mainConf, goldenMainPath)
+				updateOrCompareGolden(t, testName, platform.OS, expectedParserConfig, parserConf, goldenParserPath)
 			}
-			// Compare the expected and actual and error out in case of diff.
-			updateOrCompareGolden(t, testName, platform.OS, expectedMainConfig, mainConf, goldenMainPath)
-			updateOrCompareGolden(t, testName, platform.OS, expectedParserConfig, parserConf, goldenParserPath)
 
 			if platform.OS == "windows" {
 				expectedOtelConfig := readFileContent(t, platform.OS, testName, goldenOtelPath, true)
 				otelConf, err := uc.GenerateOtelConfig(platform.InfoStat)
 				if err != nil {
-					t.Fatalf("GenerateOtelConfig got %v", err)
+					t.Errorf("GenerateOtelConfig got %v", err)
+				} else {
+					// Compare the expected and actual and error out in case of diff.
+					updateOrCompareGolden(t, testName, platform.OS, expectedOtelConfig, otelConf, goldenOtelPath)
 				}
-				// Compare the expected and actual and error out in case of diff.
-				updateOrCompareGolden(t, testName, platform.OS, expectedOtelConfig, otelConf, goldenOtelPath)
 			} else {
 				expectedCollectdConfig := readFileContent(t, platform.OS, testName, goldenCollectdPath, true)
 				collectdConf, err := uc.GenerateCollectdConfig(platform.defaultLogsDir)
 				if err != nil {
-					t.Fatalf("GenerateCollectdConfig got %v", err)
+					t.Errorf("GenerateCollectdConfig got %v", err)
+				} else {
+					// Compare the expected and actual and error out in case of diff.
+					updateOrCompareGolden(t, testName, platform.OS, expectedCollectdConfig, collectdConf, goldenCollectdPath)
 				}
-				// Compare the expected and actual and error out in case of diff.
-				updateOrCompareGolden(t, testName, platform.OS, expectedCollectdConfig, collectdConf, goldenCollectdPath)
 			}
 		})
 	}
 }
 
-func readFileContent(t *testing.T, goos, testName, filePathFormat string, respectGolden bool) []byte {
+func readFileContent(t *testing.T, goos string, testName string, filePathFormat string, respectGolden bool) []byte {
```

Comment:
Just curious, was there a reason to order them [`goos`, `testName`] here, but [`testName`, `goos`] in `updateOrCompareGolden`? FWIW, it's probably better to keep `t` and `testName` next to each other…

Reply:
No, I just didn't compare the signatures. Will update.